nixos/modules/services/web-apps/geoserver.nix

```diff
@@ -0,0 +1,76 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.geoserver;
+
+in
+{
+  options = {
+    services.geoserver = {
+      enable = mkEnableOption "Geoserver service";
+
+      package = lib.mkPackageOption pkgs "geoserver" { };
+
+      user = mkOption {
+        type = types.str;
+        default = "geoserver";
+        description = "User which runs Geoserver.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "geoserver";
+        description = "Group which runs Geoserver.";
+      };
+
+      datadir = mkOption {
+        type = types.str;
+        default = "/var/lib/geoserver";
+        description = "The data directory where data & its configuration is stored.";
+      };
+
+      jvmOpts = mkOption {
+        type = types.lines;
+        default = "";
+        description = "Any options passed to the JVM.";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    users.users.${cfg.user} = {
+      isSystemUser = true;
+      inherit (cfg) group;
+      home = "${cfg.datadir}";
+      createHome = true;
+    };
+
+    users.groups.${cfg.group} = { };
+
+    systemd.services.geoserver = {
+      description = "Geoserver";
+
+      wantedBy = [ "multi-user.target" ];
+
+      environment = {
+        GEOSERVER_HOME = "${cfg.package}/share/geoserver";
+        GEOSERVER_DATA_DIR = "${cfg.datadir}";
+        JAVA_OPTS = "${cfg.jvmOpts}";
+      };
+
+      serviceConfig = {
+        User = cfg.user;
```

Comment:
* I suggest to use systemd's `DynamicUser` instead of real user/group.

See for example here: https://github.com/NixOS/nixpkgs/blob/d51f8579f03a76718d643dd184ae964e6f0dd751/nixos/modules/services/matrix/matrix-sliding-sync.nix#L96

* It is also worth to consider to use `StateDirectory` instead of manual `datadir` configuration.

Reply:
Ah, I considered both but refrained from it because I thought people might want to copy stuff into the data dir.
But fine with me, I'll adapt the service.

Reply:
@rolfschr , are you still interested in finishing this PR ?